http/src/test/java/io/micronaut/http/ssl/PemParserTest.java

```diff
@@ -0,0 +1,179 @@
+package io.micronaut.http.ssl;
+
+import org.junit.jupiter.api.Test;
+
+import java.math.BigInteger;
+import java.security.interfaces.ECPrivateKey;
+import java.security.interfaces.RSAPrivateKey;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+
+class PemParserTest {
```

Comment:
could you add more tests for the error conditions and assert we are producing good errors for invalid config